### PR TITLE
[WIP] Tests for new SameDiff ops

### PR DESCRIPTION
**Pull request by farizrahman4u**
Related:

#2730
#2736 

* Note: Not tested at all.. some issues when building from source, I am stuck with just snapshots for now.

### CODE REVIEW

File: nd4j-backends/nd4j-tests/src/test/java/org/nd4j/autodiff/samediff/SameDiffTests.java [hunk 1]
```diff
@@ -276,10 +277,73 @@ public void testTranspose() {
...
+        assertEquals(expected1, result1.eval());
+        assertEquals(expected2, result2.eval());
+
+    }
 
+    @Test
+    public void testGatherNd() {
+        SameDiff sameDiff = SameDiff.create();
+        INDArray arr1 = Transforms.sigmoid(Nd4j.linspace(1, 24, 24)).reshape(2, 3, 4);
+        INDArray arr2 = Nd4j.create(new int[]{1, 2, 3, 0, 1, 3, 1, 0, 2}, new int[]{3, 3});
```
Comment (maxpumperla) [inline nd4j-backends/nd4j-tests/src/test/java/org/nd4j/autodiff/samediff/SameDiffTests.java]: this won't work. ND4J will think the first int array has shape info. 0 is not allowed. etc.

**Comment by maxpumperla** [comment]
also @farizrahman4u if you could just work from a branch of nd4j from `deeplearning4j` domain, it'd be much easier for me to just check out your work and run the tests (without cloning your fork).

**Comment by maxpumperla** [comment]
@farizrahman4u where are we with this? let me know if you need help with something

**Comment by farizrahman4u** [comment]
@maxpumperla Gather and UnStack are failing. There are older ops failing as well. Let's merge these tests now and fix ops in a different PR.

**Pull request merged**